File: 605/Can_Place_Flowers.hs
import Data.List
import Data.Bool

-- define helper that keeps left val
helpMax :: [Int] -> Int -> Int
helpMax [x] left | left == 0 && x == 0 = 1
                | otherwise = 0
helpMax (x:right:xs) left | (x == 0 && left == 0 && right == 0) = (helpMax (right:xs) 1) + 1
                        | (x == 0 && left == 1) = helpMax (right:xs) 0
                        | otherwise = helpMax (right:xs) 1


-- handle user input 
findSoln :: [Int] -> Int -> Bool
findSoln lst num | helpMax lst 0 < num = False
                | otherwise = True

File: 605/Can_Place_Flowers.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    bool canPlaceFlowers(vector<int> &flowerbed, int n)
    {
        // [1,0,0,0,1]
        int num = 0;
        int left = 0;

        for (int i = 0; i < flowerbed.size(); i++)
        {
            if (n == num)
            {
                break;
                return true;
            }
            if (flowerbed[i] == 1)
            {
                left = 1;
                continue;
            }
            if (i + 1 != flowerbed.size())
            {
                if (left == 0 && flowerbed[i + 1] == 0)
                {
                    num += 1;
                    left = 1;
                    continue;
                }
                left = 0;
                continue;
            }
            if (left == 0)
            {
                num += 1;
                break;
            }
        }
        if (n > num)
        {
            return false;
        }
        return true;
    }
};